georasters/georasters.py

```diff
@@ -114,24 +114,25 @@ def aggregate(raster, ndv, block_size):
     return raster2
 
 # Function to write a new file.
-def create_geotiff(name, Array, driver, ndv, xsize, ysize, geot, projection, datatype, band=1):
+def create_geotiff(name, Array, driver, ndv, xsize, ysize, geot, projection, datatype, band=1, **kwargs):
     '''
-    Creates new geotiff from array
+    Creates new geotiff from array.
     '''
     if isinstance(datatype, np.int) == False:
         if datatype.startswith('gdal.GDT_') == False:
             datatype = eval('gdal.GDT_'+datatype)
     newfilename = name+'.tif'
     # Set nans to the original No Data Value
-    Array[np.isnan(Array)] = ndv
+    if len(Array[np.isnan(Array)]) > 0:
+        Array[np.isnan(Array)] = ndv
     # Set up the dataset
-    DataSet = driver.Create(newfilename, xsize, ysize, 1, datatype)
+    DataSet = driver.Create(newfilename, xsize, ysize, 1, datatype, **kwargs)
     # the '1' is for band 1.
     DataSet.SetGeoTransform(geot)
     DataSet.SetProjection(projection.ExportToWkt())
     # Write the array
     DataSet.GetRasterBand(band).WriteArray(Array)
-    DataSet.GetRasterBand(band).SetNoDataValue(ndv)
+    DataSet.GetRasterBand(band).SetNoDataValue(np.float64(ndv))
```

Comment:
Doesn't forcing a ``np.float64(ndv)`` force any dataset to become float, even the ones that are integers and so increasing file size?